### DISCUSSION TITLE
How to implement a series elastic actuator and a parallel elastic actuator

### DISCUSSION THREAD

**Discussion by haru425** [google-deepmind/mujoco#2140] [Asking for Help]
### Intro

Hi!

I am a graduate student at Tokyo Institue of Technology. I use MuJoCo for my research on soft robotics.

### My setup

mujoco version is 3.2.3.

And I have written the following xml.

```
<mujoco model="simple_robot">
    <compiler coordinate="local" angle="degree" />
    <option timestep="0.01" />
    <asset>
        <texture type="skybox" builtin="gradient" rgb1="0.3 0.5 0.7" rgb2="0 0 0" width="512" height="3072"/>
        <texture type="2d" name="groundplane" builtin="checker" mark="edge" rgb1="0.2 0.3 0.4" rgb2="0.1 0.2 0.3"
            markrgb="0.8 0.8 0.8" width="300" height="300"/>
        <material name="groundplane" texture="groundplane" texuniform="true" texrepeat="5 5" reflectance="0.2"/>
    </asset>

    <worldbody>
        <light mode="targetbodycom" target="base" pos="0 0 10" diffuse="50 50 50" specular="100 100 100"/>
        <geom name="floor" size="0 0 0.01" type="plane" material="groundplane"/>

        <body name="base" pos="0 0 0.1">
            <geom type="box" size="0.5 0.5 0.1" rgba="1 0 0 0.5" mass="5"/>
            <joint type="free" name="base_joint"/>
            <body name="arm1" pos="0.39 0 0.40" euler="0 45 0">
                <geom type="capsule" size="0.05 0.50" rgba="0 1 0 0.5"/>
                <body name="arm2" pos="0 0 1" euler="0 0 0">
                    <geom type="capsule" size="0.05 0.50" rgba="0 0 1 0.5"/>
                    <joint type="hinge" name="joint1" axis="0 1 0" pos="0 0 -0.5" range="-180 180"/>
                </body>
            </body>
        </body>
    </worldbody>

    <actuator>
        <motor joint="joint1" gear="10" />
    </actuator>

</mujoco>
```

### My question

I am considering implementing a series elastic actuator and a parallel elastic actuator for `joint1` in the above XML file. How can I implement each of these? 

Does modifying it as shown below represent a parallel elastic actuator?
```
<joint type="hinge" name="joint1" axis="0 1 0" pos="0 0 -0.5" range="-180 180" stiffness="50"/>
```

Also, I referred to https://github.com/google-deepmind/mujoco/discussions/226, does modifying it as shown below represent a series elastic actuator? is this modification is enough?
```
<joint type="hinge" name="joint1" axis="0 1 0" pos="0 0 -0.5" range="-180 180" armature="1"/>
```

I am a beginner with MuJoCo, so I don't understand it well.
Thank you.

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
1. What is a series elastic actuator?
2. What is a parallel elastic actuator
